lib/stripe/credit_card_cloner.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# Here we clone
+#   - a card (card_*) or payment_method (pm_*) stored (in a customer) in a platform account
+# into
+#   - a payment method (pm_*) (in a new customer) in a connected account
+#
+# This is required when using the Stripe Payment Intents API:
+#   - the customer and payment methods are stored in the platform account
+#       so that they can be re-used across multiple sellers
+#   - when a card needs to be charged, we need to create it in the seller's stripe account
+#
+# We are doing this process every time the card is charged:
+#   - this means that, if the customer uses the same card on the same seller multiple times,
+#       the card will be created multiple times on the seller's account
```

Comment:
I've got a couple of questions, just out of interest:

- What are the consequences of the same card being created multiple times on the seller's account?
- If a card already exists on the seller's account, can't we retrieve it using the Stripe API instead of recreating it?

Reply:
1. until now what was created every time on the seller's account was the token, there were as many tokens as payments on these accounts. the stripe dashboard doesnt show the tokens I think. Now we are doing the same for cards, this appears in the dashboard on the customers tab, each time there is a payment there will be a new customer, even if the card has been previously used. I dont think this is a big problem, it's just a detail in the stripe UI for the sellers.
2. yes we can, it's just more code and more work to get that done. I dont think there's much value in implementing that.